src/fsharp/ConstraintSolver.fs

```diff
@@ -1933,8 +1933,8 @@ and CanMemberSigsMatchUpToCheck
         // - Never take into account return type information for constructors 
         match reqdRetTyOpt with 
         | None -> CompleteD 
-        | Some _  when minfo.IsConstructor -> CompleteD 
-        | Some _  when not alwaysCheckReturn && isNil unnamedCalledOutArgs -> CompleteD 
+        | _ when minfo.IsConstructor -> CompleteD 
+        | _ when not alwaysCheckReturn && isNil unnamedCalledOutArgs -> CompleteD 
```

Comment:
I actually prefer leaving the ``Some _`` in in this case, I don't have to compute it for myself and it makes the clauses reorderable